cli/cmd/check.go

```diff
@@ -43,6 +46,7 @@ func (options *checkOptions) nonConfigFlagSet() *pflag.FlagSet {
 	flags.StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace to use for --proxy checks (default: all namespaces)")
 	flags.BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
 	flags.BoolVar(&options.dataPlaneOnly, "proxy", options.dataPlaneOnly, "Only run data-plane checks, to determine if the data plane is healthy")
+	flags.StringVarP(&options.output, "output", "o", options.output, "Output format. One of: basic, json")
```

Comment:
`One of: table, json` ;)

Reply:
@siggy Does this flag need to part of `linkerd check config`?

```
$ bin/go-run cli check config -o json
Error: unknown shorthand flag: 'o' in -o
Usage:
  linkerd check config [flags]

Examples:
  # Check that the Linkerd cluster-wide resource are installed correctly
  linkerd check config
```

Reply:
@ihcsim nice catch!

@alenkacz per Ivan's comment, we recently added a `linkerd check config` subcommand, which should also benefit from your json work, you can make it available to the subcommand with:
```diff
diff --git a/cli/cmd/check.go b/cli/cmd/check.go
index 0b70d87e..a0744bc1 100644
--- a/cli/cmd/check.go
+++ b/cli/cmd/check.go
@@ -46,7 +46,6 @@ func (options *checkOptions) nonConfigFlagSet() *pflag.FlagSet {
        flags.StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace to use for --proxy checks (default: all namespaces)")
        flags.BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
        flags.BoolVar(&options.dataPlaneOnly, "proxy", options.dataPlaneOnly, "Only run data-plane checks, to determine if the data plane is healthy")
-       flags.StringVarP(&options.output, "output", "o", options.output, "Output format. One of: basic, json")

        return flags
 }
@@ -56,6 +55,7 @@ func (options *checkOptions) checkFlagSet() *pflag.FlagSet {
        flags := pflag.NewFlagSet("check", pflag.ExitOnError)

        flags.StringVar(&options.versionOverride, "expected-version", options.versionOverride, "Overrides the version used when checking if Linkerd is running the latest version (mostly for testing)")
+       flags.StringVarP(&options.output, "output", "o", options.output, "Output format. One of: basic, json")
        flags.DurationVar(&options.wait, "wait", options.wait, "Maximum allowed time for all tests to pass")

        return flags
```